app/services/messages.js

```diff
@@ -1,30 +1,90 @@
 import Ember from "ember";
 const { getOwner } = Ember;
+import AjaxPromise from "stock/utils/ajax-promise";
 
 export default Ember.Service.extend({
   logger: Ember.inject.service(),
   session: Ember.inject.service(),
+  store: Ember.inject.service(),
 
-  markRead: function (message) {
+  unreadMessageCount: 0,
+
+  fetchUnreadMessageCount() {
+    return this._queryMessages("unread", 1, 1)
+      .then(data => {
+        const count = data.meta && data.meta.total_count;
+        this.set("unreadMessageCount", count || 0);
+      })
+      .catch(e => this._onError(e));
+  },
+
+  _queryMessages(state, page, perPage) {
+    return new AjaxPromise("/messages", "GET", this.get("session.authToken"), {
+      state: state,
+      scope: "order",
+      page: page,
+      per_page: perPage
+    });
+  },
+
+  markRead: function(message) {
     if (message.get("isUnread")) {
       var adapter = getOwner(this).lookup("adapter:application");
       var url = adapter.buildURL("message", message.id) + "/mark_read";
-      adapter.ajax(url, "PUT")
+      adapter
+        .ajax(url, "PUT")
         .then(data => {
           delete data.message.id;
           message.setProperties(data.message);
+          this._decrementCount();
         })
         .catch(error => this.get("logger").error(error));
     }
   },
 
+  markAllRead() {
```

Comment:
@swatijadhav I think this endpoint might also mark all the Admin notifications as read 🤔 , which we don't want.

But I believe this endpoint takes in a `"context"` param, which you can set to `"order"`

Reply:
I guess we need to check that Admin does the same

Reply:
Thanks for pointing out this one. Changes done!